pkgs/development/compilers/inklecate/default.nix

```
@@ -1,54 +1,42 @@
-{ lib, stdenv, fetchurl, unzip, makeWrapper }:
-
-stdenv.mkDerivation rec {
+{ lib
+, stdenv
+, autoPatchelfHook
+, buildDotnetModule
+, dotnetCorePackages
+, fetchFromGitHub
+}:
+
+buildDotnetModule rec {
   pname = "inklecate";
   version = "1.0.0";
 
-  src =
-    if stdenv.isLinux then
-      fetchurl {
-        url    = "https://github.com/inkle/ink/releases/download/v${version}/inklecate_linux.zip";
-        sha256 = "6e17db766222998ba0ae5a5da9857e34896e683b9ec42fad528c3f8bea7398ea";
-        name   = "${pname}-${version}";
-      }
-    else if stdenv.isDarwin then
-      fetchurl {
-        url    = "https://github.com/inkle/ink/releases/download/v${version}/inklecate_mac.zip";
-        sha256 = "b6f4dd1f95c180637ce193dbb5fa6d59aeafe49a2121a05b7822e6cbbaa6931f";
-        name   = "${pname}-${version}";
-      }
-    else throw "Not supported on ${stdenv.hostPlatform.system}.";
-
-  # Work around the "unpacker appears to have produced no directories"
-  # case that happens when the archive doesn't have a subdirectory.
-  setSourceRoot = "sourceRoot=$(pwd)";
-
-  nativeBuildInputs = [ unzip makeWrapper ];
-
-  unpackPhase = ''
-    unzip -qq -j $src -d $pname-$version
-
-    rm $pname-$version/ink-engine-runtime.dll
-    rm $pname-$version/ink_compiler.dll
-  '';
+  src = fetchFromGitHub {
+    owner = "inkle";
+    repo = "ink";
+    rev = "v${version}";
+    sha256 = "00lagmwsbxap5mgnw4gndpavmv3xsgincdaq1zvw7fkc3vn3pxqc";
+  };
 
-  installPhase = ''
-    mkdir -p $out/bin/
+  nativeBuildInputs = [ autoPatchelfHook ];
+  buildInputs = [ stdenv.cc.cc.lib ];
 
-    cp $pname-$version/inklecate $out/bin/inklecate
-  '';
+  projectFile = "inklecate/inklecate.csproj";
+  nugetDeps = if stdenv.isDarwin then ./deps-darwin.nix else ./deps-linux.nix;
+  executables = [ "inklecate" ];
 
+  dotnet-runtime = dotnetCorePackages.runtime_3_1;
+  dotnet-sdk = dotnetCorePackages.sdk_6_0;
 
   meta = with lib; {
-    description     = "Compiler for ink, inkle's scripting language";
+    description = "Compiler for ink, inkle's scripting language";
     longDescription = ''
       Inklecate is a command-line compiler for ink, inkle's open source
       scripting language for writing interactive narrative
-      '';
-    homepage        = "https://www.inklestudios.com/ink/";
-    downloadPage    = "https://github.com/inkle/ink/releases";
-    license         = licenses.mit;
-    platforms       = platforms.unix;
-    maintainers     = with maintainers; [ shreerammodi ];
+    '';
+    homepage = "https://www.inklestudios.com/ink/";
+    downloadPage = "https://github.com/inkle/ink/";
+    license = licenses.mit;
+    platforms = platforms.unix;
```

```suggestion
    platforms = platforms.unix;
    badPlatforms = platforms.aarch64;
```

Since it seems to fail on both aarch64-Darwin and aarch64-Linux

---

Can you provide a log with the error? Some dotnet programs need specific nuget deps files per platform (as is the case in this PR with operating systems)

---

In the [ArchiSteamFarm package](https://github.com/legendofmiracles/nixpkgs/blob/7d1d2fe2f8c930dac1ea34b34b26979e4c5658fc/pkgs/applications/misc/ArchiSteamFarm/default.nix#L26) I had the same error, fixed it by having separate dependency files for each platform. 

Maybe the [update script](https://github.com/legendofmiracles/nixpkgs/blob/7d1d2fe2f8c930dac1ea34b34b26979e4c5658fc/pkgs/applications/misc/ArchiSteamFarm/updater.sh) I wrote can help with generating it for all platforms, even when the machine isn't native to it, through binfmt. 

In total, that would make for 4 dependency files here :/

---

Yes :/

@IvarWithoutBones do you know of a similar situation for another package using `buildDotnetModule`?

Maybe it's possible to write just one deps.nix where only `Microsoft.NETCore.App.Host.*` is conditional on the platform?

---

It seems to me like that ought to be possible, since it's just the one dependency that seems to differ from platform to platform. But should we go ahead and merge this PR in the meantime with the suggested change? The official binary releases of inklecate don't target x64, so I don't think there would be any regressions from it.

---

Indeed if there is no upstream support for ARM x64 platforms anyway I would be ok with that.
I'll commit my initial suggestion to your branch.

---

I've only seen this before with `ArchiSteamFarm` as @legendofmiracles mentioned, not sure what a proper fix would be. We could change the way we pin nuget dependencies as suggested, but I'm not sure how much sense that makes considering all other packages function as expected on different platforms. 

I'll look into this in the future, but I should mention that it currently has a fairly low priority for me because of the small amount of packages this influences. I think the solution used now is fine, although a bit ugly.